services/multitenant/legacy.go

```diff
@@ -0,0 +1,19 @@
+package multitenant
+
+import (
+	"time"
+)
+
+type Legacy struct {
```

Comment:
It hadn't crossed my mind that there is an easy way to implement the decorator pattern in go using embedded structs, nice! 👍 

If I had to grumble about something, this might be naming, but I won't do it, since I haven't even reached to a consensus with myself about naming stuff in go yet!

This however appears to be a good opportunity to express my conviction, again, about `MultitenantI` interface, which is that only `GetRouterPickupJobs` method should be part of it, everything else appears to be part of something else.